fluss-common/src/main/java/com/alibaba/fluss/metadata/ResolvedPartitionSpec.java

```diff
@@ -151,4 +152,28 @@ private static List<String> getReorderedPartitionValues(
                 partitionKey -> reOrderedPartitionValues.add(partitionSpecMap.get(partitionKey)));
         return reOrderedPartitionValues;
     }
+
+    public boolean contains(ResolvedPartitionSpec other) {
+        List<String> otherPartitionKeys = other.getPartitionKeys();
+        List<String> otherPartitionValues = other.getPartitionValues();
+
+        List<String> expectedPartitionValues = new ArrayList<>();
+        for (int i = 0; i < otherPartitionKeys.size(); i++) {
```

Comment:
Change for loop to enhanced for loop as `for (String otherPartitionKey : otherPartitionKeys)`